rope_.d

```diff
@@ -74,19 +74,19 @@ template Rope(T,S=void)if(is(S==void) || is(typeof((S a,S b)=>a.combine(b)))){
 		static if(segtree){
 			@property value(){ return isArray()?tree?tree.value:S.init:rope.value; }
 			private this(T[] array, SliceSegTree* tree){
-				if(array==[]) return;
 				this.array = array;
+				if(array==[]) return;
 				this.tree = tree;
 			}
 		}
 
 		private this(RopeImpl* rope){ this.rope = rope; }
 		invariant(){ assert(cast(void*)rope is array.ptr); }
-		private union{
+		/*private*/ union{
```

Comment:
The problem is that `array` is accessed from semantic.d. I just tried to replace these accesses by a public property function. Testing `test/ttuple` still crashes for me (with the new compiler, though).

I see the stack overflow in _doAnalyze where it repeats this chain of references: ExpTuple.exprs[i] -> BinaryExp!68.e1 -> TmpVarExp.sym -> ExpTuple.exprs. I was rather expecting the rename of `init` to  `init_` has changed the member iteration somehow.